SwiftLint 0.62.1 fails to build in release mode due to TestHelpers using @testable import

### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

SwiftLint 0.62.1 fails to build when using swift build -c release without specifying a target. The build fails because the TestHelpers target uses @testable import SwiftLintFramework, but SwiftLintFramework is not compiled with testability enabled in release builds.

### Reproduction Steps

```bash
# Clone SwiftLint 0.62.1
git clone --depth 1 --branch 0.62.1 https://github.com/realm/SwiftLint.git
cd SwiftLint
```

```bash
# Try building in release mode (fails)
swift build -c release
```

### Expected Behavior
The build should succeed, or at minimum, test-only targets should not be built in release configuration.

### Actual Behavior

Build fails with:

```bash
  [152/153] Compiling TestHelpers RuleDescription+Examples.swift
  /path/to/Tests/TestHelpers/TestHelpers.swift:5:18: error: module 'SwiftLintFramework' was not compiled for testing
    3 | import XCTest
    4 |
    5 | @testable import SwiftLintFramework
      |                  `- error: module 'SwiftLintFramework' was not compiled for testing
    6 |
    7 | // swiftlint:disable file_length
```

### Workaround

Building with a specific target works:

```bash
swift build -c release --target swiftlint
```

### Possible Root Cause

In Package.swift, TestHelpers is defined as a regular .target() instead of .testTarget():

```bash
  .target(
      name: "TestHelpers",
      dependencies: [
          "SwiftLintFramework"
      ],
      path: "Tests/TestHelpers",
      exclude: [
          "BUILD",
      ],
      swiftSettings: swiftFeatures + strictConcurrency
  ),
```

This causes it to be built during all builds, including release builds. However, it contains @testable import SwiftLintFramework, which requires the framework to be compiled with testability enabled—something that doesn't happen in release builds.

### Impact
This issue affects users who:
- Use tools that build SwiftLint from source without specifying --target
- Use pre-commit hooks (pre-commit's Swift support doesn't allow specifying build targets)
- Integrate SwiftLint into custom build systems

### Environment
- SwiftLint version: 0.62.1
- Swift version: 6.0+ (though affects all versions)
- Platform: macOS (likely affects Linux as well)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -12,7 +12,10 @@
 
 ### Enhancements
 
-* None.
+* Declare `TestHelpers` as a test target in `Package.swift` to avoid building
+  and linking it in release mode.  
+  [SimplyDanny](https://github.com/SimplyDanny)
+  [#6299](https://github.com/realm/SwiftLint/issues/6299)
 
 ### Bug Fixes
 

--- a/Package.swift
+++ b/Package.swift
@@ -216,7 +216,7 @@ let package = Package(
             ],
             swiftSettings: swiftFeatures + strictConcurrency
         ),
-        .target(
+        .testTarget(
             name: "TestHelpers",
             dependencies: [
                 "SwiftLintFramework"

